contracts/croncat-manager/src/helpers.rs

```diff
@@ -603,3 +546,69 @@ pub fn create_task_completed_msg(
 
     Ok(execute)
 }
+
+pub(crate) trait AssertCaller {
+    fn assert_caller_is_agent_contract(
+        &self,
+        deps_queries: &QuerierWrapper<Empty>,
+        config: &Config,
+    ) -> Result<(), ContractError>;
+
+    fn assert_caller_is_tasks_contract(
+        &self,
+        deps_queries: &QuerierWrapper<Empty>,
+        config: &Config,
+    ) -> Result<(), ContractError>;
+
+    fn assert_caller_is_task_owner(
+        &self,
+        querier: &QuerierWrapper,
+        tasks_addr: &Addr,
+        task_hash: &str,
+    ) -> Result<(), ContractError>;
+}
+
+impl AssertCaller for Addr {
+    fn assert_caller_is_agent_contract(
+        &self,
+        deps_queries: &QuerierWrapper<Empty>,
+        config: &Config,
+    ) -> Result<(), ContractError> {
+        let addr = get_agents_addr(deps_queries, config)?;
+        if addr != *self {
+            return Err(ContractError::Unauthorized {});
+        }
+        Ok(())
```

Comment:
```suggestion
        if addr != *self {
            Err(ContractError::Unauthorized {})
        } else {
            Ok(())
        }
```
looks cleaner this way right?